From 3f40c958104c4a83a32ad303cd7153f79cb7a6a8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Wed, 11 Jan 2017 14:45:19 -0500
Subject: [PATCH] shell-completion: redirect all errors from systemctl to
 /dev/null

Completion scripts should not generate errors, ever.

https://bugzilla.redhat.com/show_bug.cgi?id=1409649
(cherry picked from commit 99171d2fdf720ebb64fee75db4177af5d048d0f9)
---
 shell-completion/bash/systemctl.in | 2 +-
 shell-completion/zsh/_systemctl.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/shell-completion/bash/systemctl.in b/shell-completion/bash/systemctl.in
index dcf71a1f51..34589e2888 100644
--- a/shell-completion/bash/systemctl.in
+++ b/shell-completion/bash/systemctl.in
@@ -19,7 +19,7 @@
 
 __systemctl() {
         local mode=$1; shift 1
-        systemctl $mode --full --no-legend "$@"
+        systemctl $mode --full --no-legend "$@" 2>/dev/null
 }
 
 __systemd_properties() {
diff --git a/shell-completion/zsh/_systemctl.in b/shell-completion/zsh/_systemctl.in
index 03a1c930b0..d77a2df74e 100644
--- a/shell-completion/zsh/_systemctl.in
+++ b/shell-completion/zsh/_systemctl.in
@@ -91,7 +91,7 @@
 
 __systemctl()
 {
-  systemctl $_sys_service_mgr --full --no-legend --no-pager "$@"
+  systemctl $_sys_service_mgr --full --no-legend --no-pager "$@" 2>/dev/null
 }
 
 
